CMakeLists.txt

```diff
@@ -561,16 +561,39 @@ find_program(CLANG_FORMAT NAMES clang-format clang-format-4.0 clang-format-3.9 c
 
 if(CLANG_FORMAT)
   message(STATUS "Looking for clang-format - found")
-  file(GLOB_RECURSE LINT_ALL_SOURCE_FILES src/*.cc include/*.h)
-  add_custom_target(
-    format
-    COMMAND
-      ${CLANG_FORMAT}
-      -style=file
-      -sort-includes
-      -i
-      ${LINT_ALL_SOURCE_FILES}
-  )
+  file(GLOB_RECURSE FORMAT_ALL_SOURCE_FILES src/*.cc include/*.h)
+  string(LENGTH "${FORMAT_ALL_SOURCE_FILES}" FORMAT_ALL_SOURCE_FILES_LEN)
+  if(WIN32 AND FORMAT_ALL_SOURCE_FILES_LEN GREATER 8192)
+    # use multiple targets to overcome cmd 8192 length limit
+    add_custom_target(format)
+
+    foreach(FORMAT_SOURCE_FILE ${FORMAT_ALL_SOURCE_FILES})
+      list(APPEND FORMAT_ADDED_TARGETS ${FORMAT_SOURCE_FILE})
+      list(LENGTH FORMAT_ADDED_TARGETS FORMAT_ADDED_TARGETS_LEN)
```

Comment:
Why not an int variable? Seems that only length of FORMAT_ADDED_TARGETS is used.